File: Programmers/py/lev1/추억 점수/solution.py
def solution(name, yearning, photo):
    dic = dict()
    for i in range(len(name)):
        dic[name[i]] = yearning[i]
    
    return [sum(dic[p_name] for p_name in n_lst if p_name in dic.keys()) for n_lst in photo]

# name = ["may", "kein", "kain", "radi"]
# yearning = [5, 10, 1, 3]
# photo = [["may", "kein", "kain", "radi"],["may", "kein", "brin", "deny"], ["kon", "kain", "may", "coni"]]
name = ["may", "kein", "kain", "radi"]
yearning = [5, 10, 1, 3]
photo = [["may"],["kein", "deny", "may"], ["kon", "coni"]]
print(solution(name, yearning, photo))

File: Programmers/java/lev1/추억 점수/Solution.java
import java.util.*;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        HashMap<String,Integer> map = new HashMap<>();
        int[] answer = new int[photo.length];
        for (int i=0 ; i<name.length; i++)
            map.put(name[i],yearning[i]);

        for( int i =0 ; i< photo.length; i++)
        {
            int point = 0 ;
            for (String person : photo[i])
                point += map.getOrDefault(person,0);
            answer[i] = point;
        }
        return answer;
    }

    public static void main(String[] args)
    {
        Solution t = new Solution();

        String[] name = {"may", "kein", "kain", "radi"};
        int[] yearning = {5, 10, 1, 3};
        String[][] photo = {{"may", "kein", "kain", "radi"},{"may", "kein", "brin", "deny"}, {"kon", "kain", "may", "coni"}};
        System.out.println(t.solution(name, yearning,photo));
    }
}